packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -352,9 +425,32 @@ export class OrderedClientElection
         if (this.isEligibleFn(client)) {
             this._eligibleCount--;
             if (this._electedClient === client) {
-                // Automatically shift to next oldest client
-                const nextClient = this.findFirstEligibleClient(this._electedClient.youngerClient);
-                this.tryElectingClient(nextClient, sequenceNumber);
+                // Removing the _electedClient. There are 2 possible cases:
+                if (this._electedParent !== client) {
+                    // 1. The _electedClient is a summarizer that we've been allowing to finish its work.
+                    // Let the _electedParent become the _electedClient so that it can start its own summarizer.
+                    if (this._electedClient.client.details.type !== summarizerClientType) {
+                        throw new UsageError("Elected client should be a summarizer client 1");
+                    }
+                    this.tryElectingClient(this._electedParent, sequenceNumber);
+                }
+                else {
+                    // 2. The _electedClient is an interactive client that has left the quorum.
+                    // Automatically shift to next oldest client.
+                    const nextClient = this.findFirstEligibleParent(this._electedParent?.youngerClient) ??
+                        this.findFirstEligibleParent(this.orderedClientCollection.oldestClient);
+                    this.tryElectingClient(nextClient, sequenceNumber);
+                }
+            }
+            else if (this._electedParent === client) {
+                // Removing the _electedParent. Shift to the next oldest one, but do not replace the _electedClient,
```

Comment:
This scenario is where the electedClient and electedParent are the same. Again, referring to the scenario in the comment, this should happen when the old summarizer has left and the electedParent is chosen as the electedClient as well. So, the next step would have been for the electedParent to spawn a summarizer client which becomes electedClient.
If this client is torn down, doesn't that mean we should select both electedParent and electedClient? I am confused by this comment here.

Reply:
This is where electedClient and electedParent are _not_ equal. If they were equal, we would have entered the "if" path. On the "else" path, the electedParent is being removed, so we're going to change electedParent _only_. That will cause SummaryManager on that client to call Summarizer.stop.

Reply:
Again, I thought this is the else to the inner if :-(.